index.js

```diff
@@ -46,24 +53,32 @@ const computeChannelId = (src, dest, sequence) => {
 }
 
 const claimFunds = async (self, amount, signature) => {
-  const txTag = randomTag()
   const tx = await self.api.preparePaymentChannelClaim(self.address, {
-    balance: new BigNumber(amount).div(1000000).toString(),
+    balance: dropsToXrp(amount),
     channel: self.incomingPaymentChannelId,
-    signature: signature,
-    publicKey: self.incomingPaymentChannel.PublicKey
+    signature: signature.toUpperCase(),
+    publicKey: self.incomingPaymentChannel.publicKey
   })
 
   const signedTx = self.api.sign(tx.txJSON, self.secret)
-  await self.api.submit(signedTx.signedTransaction)
+  debug('submitting claim transaction ', tx)
+  const {resultCode, resultMessage} = await self.api.submit(signedTx.signedTransaction)
+  if (resultCode !== 'tesSUCCESS') {
+    debug('Error submitting claim: ', resultMessage)
+    throw new Error('Could not claim funds: ', resultMessage)
+  }
 
   return new Promise((resolve) => {
-    handleTransaction (ev) {
-      if (ev.transaction.SourceTag !== txTag) return
+    const handleTransaction = function (ev) {
       if (ev.transaction.Account !== self.address) return
+      if (ev.transaction.Channel !== self.incomingPaymentChannelId) return
+      if (ev.transaction.Balance !== dropsToXrp(amount)) return
```

Comment:
Is it possible to put a source tag on this transaction? That's kinda the standard way to make sure it's the same one that you sent

Reply:
The [claim](https://github.com/ripple/ripple-lib/blob/a9e5ac6290ef2fa17fadb34e808e4f5080d7ba32/docs/index.md#payment-channel-claim) passed to `preparePaymentChannelClaim` does not have a source tag. Looks like `sourceTag` is only supported for the pay chan create transaction.